### ISSUE TITLE
Contributors needed to maintain this project

### ISSUE THREAD

**Issue by tpyo** [tpyo/amazon-s3-php-class#102]
Unfortunately I am unable to continue work on this project.  If you are interested in becoming a contributor please leave a comment below.

**Comment by jimaek** [comment]
I would like to try to maintain it. What is needed for this?
I am using the library at a few personal projects myself.

**Comment by EricP** [comment]
I've gotten a lot of great use out of this class, and I've very much appreciated its simplicity of implementation. But because it's outdated I've moved to the AWS SDK (v 3.x http://docs.aws.amazon.com/aws-sdk-php/v3/guide/). 

At first, I thought it was going to be bloatware and complex. But what I've done is made a simple/light, static wrapper class around just the SDK s3 functionality ("use Aws\S3\S3Client"). The total size of the loaded AWS files is 78,928 KB. In that sense it's not bloatware at all. Not much bigger than this class...and always maintained by the AWS staff.

I may need to adjust/modify my wrapper class, but that's easy.

**Comment by robertotremonti** [comment]
@EricP Could you share your S3 "standalone" version?
I can't distribute a 1000+ files project just to get pre-signed S3 file URLs.
This class is great but unfortunately not maintained anymore.

**Comment by EricP** [comment]
Sure. Although this is VERY rough/static class for proof of concept. However, it may be a starting point for you.

http://shepardclient.s3.amazonaws.com/s3/s3.wrapper.php

If you're so inclined, feel free to repurpose/clean up and start a repository. Right now, it does a few basic things. One of them is getting the pre-signed URL:

s3::geturl($bucket,$key,$expires);

**Comment by tpyo** [comment]
@jimaek It's difficult to process pull requests without a test suite, I think that would be a good place to start if you're up for it - it would help clear up the backlog and help bring you up to speed.  We can chat about it more and how I can help on GTalk/Skype if you like, PM me your details.

@robertotremonti I started this when Amazon didn't have a PHP S3 SDK and expected a lot of folks to move to the official SDK after its release, but it seemed to serve as a good lightweight alternative so I've tried to keep it maintained, but I don't have enough time to work on it - so without new maintainers it won't be updated much.

**Comment by EricP** [comment]
@tpyo Indeed, this class has served the PHP community brilliantly prior to the official AWS SDK and through all of the early SDK issues/challenges. It has been personally invaluable to me. Kudos!!

**Comment by robertotremonti** [comment]
Thank you, @EricP.
